docs/deployment/methods/git.md

```diff
@@ -142,7 +142,7 @@ If the specified image already exists on the Dokku host, it will not be pulled a
 Triggering a build with the same arguments multiple times will result in Dokku exiting `0` early as there will be no changes detected. If the image tag is reused but the underlying image is different, it is recommended to use the image digest instead of the tag. This can be retrieved via the following command:
 
 ```shell
-docker inspect --format='{{index .RepoDigests 0}}' $IMAGE_NAME
+docker images --no-trunc --quiet $IMAGE_NAME
```

Comment:
Can you actually check what version of docker you are running? And maybe check what the output of `docker inspect` is/post it so we can verify it. This should for sure work...

https://stackoverflow.com/a/33511811

Reply:
```
Docker version 24.0.2, build cb74dfc
```

```
docker inspect dokku/app:latest
[
    {
        "Id": "sha256:719883adaf7071798ad6de2bcecbbc7113ac07653e7013f5cd1bc0349c24678e",
        "RepoTags": [
            "dokku/app:latest"
        ],
        "RepoDigests": [],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2023-06-22T00:48:12.24723248-04:00",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "NODE_VERSION=18.16.1",
                "YARN_VERSION=1.22.19"
            ],
            "Cmd": [
                "/bin/sh",
                "-c",
                "pnpm start"
            ],
            "ArgsEscaped": true,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "/app",
            "Entrypoint": [
                "docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": null
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 411919357,
        "VirtualSize": 411919357,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/ju91zz52h366l01c5reg99fzh/diff:/var/lib/docker/overlay2/2flktu4vedxo4ytmv4780ejih/diff:/var/lib/docker/overlay2/u4s01ww6u1s2r79sc6aj585mr/diff:/var/lib/docker/overlay2/peqmih9kozryn1n2i4cqjrcm4/diff:/var/lib/docker/overlay2/q1v4twbdnw2lwz9z8qw3hndr6/diff:/var/lib/docker/overlay2/yfnx3kfe60sf5fti509t399tp/diff:/var/lib/docker/overlay2/6usxajk0omx8yhegg84hrdpw1/diff:/var/lib/docker/overlay2/mov38wo1k5s9nrow7srj4yjaf/diff:/var/lib/docker/overlay2/k97sjjws9muojg9q0i7qq4ac1/diff:/var/lib/docker/overlay2/x8796sra298uwg9m6oio0dlz8/diff:/var/lib/docker/overlay2/ln8fdvz6n41fmd33dnohu5baf/diff:/var/lib/docker/overlay2/yc7rtu5uuw6n8p1bq5ic6vikq/diff:/var/lib/docker/overlay2/zow60ru5u8ho3804d94kujarg/diff:/var/lib/docker/overlay2/b740e99218909f1a768c61ff5d67963eb46d04e5ff60b3badd41932b87f89716/diff:/var/lib/docker/overlay2/2fa09c09631573ae5234fa8856468478f8898555d343c3a60374d935abe13f4b/diff:/var/lib/docker/overlay2/04f9fe5c95a6af5c026f3f0f0cdf1e2235d972a6fd906e0700ec489bd42e99fd/diff:/var/lib/docker/overlay2/e89eabc4e91da9b6ad84bb4336caf242b185095f1e065050c326810c5c67e971/diff",
                "MergedDir": "/var/lib/docker/overlay2/q9vxmbonit3h4e7xiliwhq9aq/merged",
                "UpperDir": "/var/lib/docker/overlay2/q9vxmbonit3h4e7xiliwhq9aq/diff",
                "WorkDir": "/var/lib/docker/overlay2/q9vxmbonit3h4e7xiliwhq9aq/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:78a822fe2a2d2c84f3de4a403188c45f623017d6a4521d23047c9fbb0801794c",
                "sha256:a0bbbabe7b80c484324ac1f67d69c1885f9bf02283bbca48b60a668142378d8c",
                "sha256:54e2f04676149dc3efdf860331124d48631f5c3cd154f62c5cb099f257d91304",
                "sha256:879b1e5603901c1bc478ce8030660df5374670d3c14ed97d5656f93e99b93a26",
                "sha256:c150cac57270e5c4f732d9c480c74375593f0bf1cf8a77c8e913ff461e17c599",
                "sha256:f089346d501961d09b982f116e5675cfede2d1f405d81d8bdf730afcebdea15b",
                "sha256:fe1f9e5fa7f1409dc0d99f300791dce9772af37d2e7686ce91c6ba2db379d831",
                "sha256:d215a5878a7fbd2ca07821e852c3c240229470b38a3264ce510c24311deecd3f",
                "sha256:941ca46498894efd232780bad1802259e878385bf6af36546016e37a339835a2",
                "sha256:58014fc33c8184e1febbeffab818729ca5803ec505761782b85730b34ae45089",
                "sha256:794d39bb315b149560332833396acdbd8b735b965abf792c05534b6b200673d3",
                "sha256:09fbd5be36990bd8a33a74a0dfe7b58f7682798ce279170f98728dab15efa026",
                "sha256:61b493f5962b03a7ca8befdc0b3161455531e60ba61c0c85294a381f37cccf8b",
                "sha256:47cdb3a7cab26b79a5722bd3a43ce1721d61c3148d49634565ae0f2cf7438ff8",
                "sha256:0854e7422752da0e70f5c7895c133efcb712598e2aaa20cdf730b075e82703d6",
                "sha256:ff44e94730853cb42b61158c6ac251acd7fcff57dfc4b64bef10456abf15c957",
                "sha256:e1cd513acebf5872fcd7ec89c9647f0566ada5381b8a4b4b4b7999dbc96790f9",
                "sha256:41ed7dacc2e5f26fc6cbd842d737711a1796a6f616e2fa71ca190a7003f8aaca"
            ]
        },
        "Metadata": {
            "LastTagTime": "2023-06-22T00:48:13.77153954-04:00"
        }
    }
]
```

Reply:
Amazing: https://stackoverflow.com/a/39812035

Basically if you haven't pushed the image, it won't have a digest 🤦🏽

Reply:
I think the solution I came up with works though, unless you think the SHA is invalid there?

`docker images --no-trunc --quiet $IMAGE_NAME`

Reply:
More info: https://windsock.io/explaining-docker-image-ids/

I'm not sure the image ID is always a valid digest.

Reply:
@josegonzalez what do you think is the best thing to do here?

Add a note saying this only works if the image was published to a registry?

Reply:
Yeah I think so. Maybe linking out to that external post is a good idea?